docs/framework/social-post-generator.mdx

```diff
@@ -2,218 +2,410 @@
 
-To rename this application and update the Header component, open the code editor and update `my_app.title` in `wf.init_state`:
+The project editor is a web application that runs on your computer and enables you to define and edit your app's user interface. Launch it by typing the following into your terminal application:
 
-```python
-wf.init_state({
-    "my_app": {
-        "title": "SOCIAL POST GENERATOR"
-    }
-})
 ```
+writer edit social-post-generator
+```
+
+You'll be presented with the URL for the project editor. Control-click the link (Command-click on macOS) to open the project editor, or copy the URL and paste it into the address bar of a browser window.
+
+You'll be presented with the project editor:
+
+![Project editor](/framework/images/tutorial/social_post/sp_gen_2b.png)
+
+You'll see the following: 
+
+- The **canvas** is in the center. It displays the app's user interface. 
+- The column on the left contains:
+  - The **Toolkit**, which contains all the UI components. You define the user interface by dragging components from the Toolkit and placing them on the canvas.
+  - The **Component tree**, which shows the arrangement of the UI components on the canvas. It's also useful for selecting items on the canvas, especially when it has a lot of UI components.
+
+It's time to build the UI!
+
+<Steps>
+  <Step title="Examine the header">
+  Select the **Header** component by clicking it — it's the component at the top, containing the title **AI STARTER** and a gray area labeled **Empty Header**.
+
+  When you click it, you'll see the **Properties** panel appear on the right side of the page. This lets you view and edit the properties of the selected component.
+
+  The first property you'll see in the panel is the **Text** property, which defines the text that appears as the header's title. It should contain the value `@{my_app.title}`. The `@{` and `}` indicate that `my_app.title` is a variable and that its contents should be the text displayed instead of the literal text "my_app.title". You'll set the value of this variable soon.
+  </Step>
+
+  <Step title="Clear the Section's default title">
+  Select the **Section** component by clicking it — it's just below the **Header** component and contains the title **Section Title** and a gray area labeled **Empty Section**.
+
+  In the **Properties** panel, clear out the value of the **Title** property. This will remove the _Section_'s default title.
+  </Step>
+
+  <Step title="Add a Text Input component">
+  The user will need a place to enter words or phrases that the app will use as the basis for generating posts and tags.
+  
+  Drag a **Text Input** component from the **Toolkit** panel on the left (it's under **Input**, and you may need to scroll down a little to find it) and into the **Section**. Sections can act as containers for other components.
```

Comment:
It might be a nice touch to callout that you can search for components

Reply:
Done!